bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/MathUtils.java

```diff
@@ -63,4 +63,28 @@ public static long remainder(long numerator, long denominator) {
             }
         }
     }
+
+    public static long addExact(long num1, long num2) {
+        long r = num1 + num2;
+        if (((num1 ^ r) & (num2 ^ r)) < 0) {
+            throw  ErrorCreator.createError(BallerinaErrorReasons.NUMBER_OVERFLOW, INT_RANGE_OVERFLOW_ERROR);
+        }
+        return r;
+    }
+
+    public static long subtractExact(long num1, long num2) {
+        long r = num1 - num2;
+        if (((num1 ^ num2) & (num1 ^ r)) < 0) {
+            throw  ErrorCreator.createError(BallerinaErrorReasons.NUMBER_OVERFLOW, INT_RANGE_OVERFLOW_ERROR);
+        }
```

Comment:
It would be good to move the common code to a new function

Reply:
As the if condition is not same as the above one, shall I not move this to a new function. I will move the same error throwing part into a new method.